socs/agents/ld_monitor/agent.py

```diff
@@ -0,0 +1,424 @@
+                    'g_timer': data_split[8],
+                    'allclear_timer': data_split[9],
+                    'faultcode': faultcode
+                }
+
+                self.data_dict.update(self.newdata_dict)
+                return self.data_dict
+
+            # disregard "alarm timers" sentence but update sentence type
+            elif param == 'WT':
+                self.newdata_dict = {'d_type': 4}
+                self.data_dict.update(self.newdata_dict)
+                return self.data_dict
+
+    def read_cycle(self):
+        """
+        In each cycle data is read and then parsed following
+        the format required to publish data to the ocs feed
+        """
+        try:
+            cycle_data = {}
+            self.read_data()
+
+            # updates time since last strike if previous strike data exists
+            if self.data_dict['time_last'] == -1.:
+                self.data_dict['tsince_last'] = -1.
+            else:
+                self.data_dict['tsince_last'] = (time.time()
+                                                 - self.data_dict['time_last'])
+
+            # parse data to ocs agent feed format
+            for key in self.data_dict:
+                cycle_data[key] = {'value': self.data_dict[key]}
+
+            if self.verbose:
+                print(cycle_data)
+            return cycle_data
+
+        except BaseException:
+            pass
+            if self.verbose:
+                print('Passing to next data iteration')
+
+
+class ld_monitorAgent:
+    """Monitor the Lightning Detector data via UDP.
+
+    Parameters
+    ----------
+    agent : OCSAgent
+        OCSAgent object which forms this Agent
+    unit : int
+    sample_interval : float
+        Time between samples in seconds.
+
+    Attributes
+    ----------
+    agent : OCSAgent
+        OCSAgent object which forms this Agent
+    take_data : bool
+        Tracks whether or not the agent is actively issuing SNMP GET commands
+        to the ibootbar. Setting to false stops sending commands.
+    log : txaio.tx.Logger
+        txaio logger object, created by the OCSAgent
+    """
+
+    def __init__(self, agent, unit=1, sample_interval=15.):
+
+        self.unit = unit
+        self.agent: ocs_agent.OCSAgent = agent
+        self.log = agent.log
+        self.lock = TimeoutLock()
+
+        self.pacemaker_freq = 1. / sample_interval
+
+        self.initialized = False
+        self.take_data = False
+
+        self.ld_monitor = None
+
+        agg_params = {
+            'frame_length': 10 * 60  # [sec]
+        }
+        self.agent.register_feed('ld_monitor',
+                                 record=True,
+                                 agg_params=agg_params,
+                                 buffer_time=0)
+
+    def _connect(self):
+        """connect()
+        Instantiates LD object and check if client is open
+        """
+        self.ld_monitor = ld_monitor(verbose=verbosity)

```

Comment:
Please uppercase the `verbosity` constant to `VERBOSITY`. (Took me a second to realize this was a constant defined at the top of the file, rather than a missing argument.)